packages/react/docs/guides/onboarding.md

```diff
@@ -0,0 +1,259 @@
+# Onboarding
+
+The purpose of this document is to help new developers quickly and easily onboard to this project and
+pick up the styles and practices we use to make this a coherent code base.
+
+## Coding practices
+
+Below you will find major items we strive to provide in all of our code and try to check for in all
+code reviews.
+
+### Prop naming conventions
+
+When naming props all names should be in camelCase, events should be prefixed with `on` (onClick), and booleans should be prefixed with `is` or `has` (isActive, hasLabel).
+
+### Localization
+
+All non-falsy translatable strings should be passed in an `i18n` prop and the defaultProps should include
+an English default. If you have multiple default, but only want to override some of them, you can use
+the `useMerged` helper hook to merge the objects together.
+
+```jsx
+// component
+const defaultProps = {
+  onClick: undefined,
+  i18n: {
+    labelText: 'button label',
+  },
+};
+
+const Button = ({ i18n, onClick, children }) => {
+  const mergedI18n = useMerged(defaultProps.i18n, i18n);
```

Comment:
Just wanna make sure I get this, here `mergedI18n` will have the default if `i18n` is empty/null, otherwise it will use `i18n`'s value?
If `i18n` contained more than one `labelText`, some empty and some with values, would the `useMerged` use the default only for the empty ones?

Reply:
I've replaced this with a better example and more comments to describe. Let me know if that clarifies it better.

Reply:
See line 39 and following

Reply:
Much clearer, thanks for elaborating